scvi/module/_mrdeconv.py

```diff
@@ -217,14 +226,21 @@ def loss(
         px_rate = generative_outputs["px_rate"]
         px_o = generative_outputs["px_o"]
         gamma = generative_outputs["gamma"]
+        v = generative_outputs["v"]
 
         reconst_loss = -NegativeBinomial(px_rate, logits=px_o).log_prob(x).sum(-1)
 
         # eta prior likelihood
         mean = torch.zeros_like(self.eta)
         scale = torch.ones_like(self.eta)
-        glo_neg_log_likelihood_prior = -Normal(mean, scale).log_prob(self.eta).sum()
-        glo_neg_log_likelihood_prior += torch.var(self.beta)
+        glo_neg_log_likelihood_prior = (
+            -1e-10 * Normal(mean, scale).log_prob(self.eta).sum()
```

Comment:
that seems extreme, what's supporting this choice?

Reply:
Up to 1e-4 it works (as it is multiplied by n_obs it's close to the old value).

Reply:
1e-4 also works (everything above leads to wrong cell type proportions). I don't have a strong reason that it helps with the three datasets to have it. 1e-4*n_obs is close to the old term. Beta_weighting is much bigger now and added as an input parameter.